src/state/accountConfig.ts

```diff
@@ -119,6 +135,15 @@ export const reducer = (
       return { ...state, earnings: [...state.earnings, action.payload] }
     case ADD_GOAL:
       return { ...state, goals: [...state.goals, action.payload] }
+    case DELETE_GOAL:
```

Comment:
lets pass only index to `createActionDeleteGoal` and use here to simplify it to the:
`return {...state, goals: state.goals.filter((goal,index) => index !== action.payload)}`

Reply:
+ also there is no case added for the edit in reducer